lib/resque/plugins/stages.rb

```diff
@@ -15,41 +15,79 @@ class Error < StandardError; end
         if singleton_class.included_modules.map(&:name).include?("Resque::Plugins::Retry")
           try_again_callback :stages_report_try_again
           give_up_callback :stages_report_giving_up
-
-          add_record_inline_failure
         else
           add_record_failure
         end
       end
 
       # rubocop:disable Metrics/BlockLength
       class_methods do
+        def perform_job(*args)
```

Comment:
Moved `perform_jobs` here so I would have a `class_name` so I could determine if the args are/should be compressed.